src/BenchmarkDotNet/Engines/CacheClearingStrategies/AllocationsCacheClearingStrategy.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace BenchmarkDotNet.Engines.CacheClearingStrategies
+{
+    internal class AllocationsCacheClearingStrategy : ICacheClearingStrategy
```

Comment:
I would rename this type to mention OS, sth like AllocationsCacheClearingStrategy**ForWindows**